### ISSUE TITLE
alltoall gloo wrapper function returns tensor list with None values which is resulting in the failure of torch.cat function

### ISSUE THREAD

**Issue by kylasa** [dmlc/dgl#4793]
## 🐛 Bug

When running the pipeline during retrieval of SHUFFLE_GLOBAL_NIDs for local edges it might happen that some processes might not have any node-id list to send to other processes. In this case the list of tensors returned from gloo wrapper call, alltoallv, will have None values. torch.cat() function fails when concatenating a list of tensors with None values in it. 

## To Reproduce
Run the pipeline with batch messages for retrieving shuffle_global_nids for edges (make sure that both the source and destination end points are included in the outgoing list). The pipeline will fail because of the failure of torch.cat function call. 

## Environment

 - DGL Version (e.g., 1.0):
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):
 - OS (e.g., Linux):
 - How you installed DGL (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:
